src/Microsoft.Azure.WebJobs.Host/Blobs/BlobPath.cs

```diff
@@ -55,16 +55,32 @@ public static BlobPath ParseAndValidate(string value, bool isContainerBinding =
             return path;
         }
 
-        public static BlobPath Parse(string value, bool isContainerBinding)
+        // throws exception if failed
+        public static string ConvertAbsUrlToContainerBlob(string blobUrl)
         {
-            BlobPath path;
-
-            if (!TryParse(value, isContainerBinding, out path))
+            try
+            {
+                Uri uri = new Uri(blobUrl); // default to UriKind.Absolute
+                return uri.LocalPath.Substring(1); // [0] is slash, same logic as ToBlobPath() in StorageAnalyticsLogEntry.cs
+            }
+            catch (Exception)
```

Comment:
>catch (Exception) [](start = 12, length = 17)

This will add an exception on the hotpath for every [Blob] binding.  Does Uri have a TryParse method or equivalent?

Reply:
I filed [https://github.com/Azure/azure-webjobs-sdk/issues/1320](https://github.com/Azure/azure-webjobs-sdk/issues/1320) to track this issue. 

---
In reply to: [135612149](https://github.com/Azure/azure-webjobs-sdk/pull/1315#discussion_r135612149) [](ancestors = 135612149)